errors.go

```diff
@@ -0,0 +1,43 @@
+package kable
+
+import (
+	"github.com/Kable-io/kable-go/internal/openapi"
+	"net/http"
+)
+
+// Error provides access to the body, status, and error on returned errors.
+type Error struct {
```

Comment:
I know this isn't immediately going to be a publicly-consumed concept, but let's still structure this as if it were so that when we do beef out our library to handle more of the API we have a way to ensure consistency across HTTP errors and SDK errors. 

I'd suggest the following
```go
type KableError struct {
	status int
	message string
        // body []byte // is this the HTTP response body wrapped up? if so, awesome. if it's something else, what is it?
}
```